data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -333,7 +337,13 @@ private Bytes48 getBytes48FromParameter(final String validatorParameter) {
               return new GetBlockHeadersResponse(
                   bellatrixEnabled ? executionOptimistic : null,
                   blockAndMetaDataList.stream()
-                      .map(blockData -> new BlockHeader(blockData.getData(), true))
+                      .map(
+                          blockData ->
+                              new BlockHeader(
+                                  blockData.getData(),
+                                  combinedChainDataClient.isCanonicalBlock(
+                                      blockData.getData().getSlot(),
+                                      blockData.getData().getRoot())))
```

Comment:
Should just use the value from the meta data here too.

Reply:
ok this is blockAndMetadata so i can understand that bit...